src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,647 @@
+                        containerOpt: null,
+                        symbol.Name,
+                        usedNames,
+                        cancellationToken).Text;
+            }
+
+            return renameTable.ToImmutableDictionary();
+        }
+
+        private bool MergeInlineContentAndVariableDeclarationArgument(
+            TInvocationSyntax calleeInvocationNode,
+            SemanticModel semanticModel,
+            ImmutableArray<IParameterSymbol> parametersWithVariableDeclarationArgument,
+            TExpressionSyntax inlineExpressionNode,
+            CancellationToken cancellationToken)
+        {
+            if (parametersWithVariableDeclarationArgument.Length == 1
+                && _syntaxFacts.IsExpressionStatement(calleeInvocationNode.Parent)
+                && semanticModel.GetOperation(inlineExpressionNode, cancellationToken) is ISimpleAssignmentOperation simpleAssignmentOperation
+                && simpleAssignmentOperation.Target is IParameterReferenceOperation parameterOperation
+                && parameterOperation.Parameter.Equals(parametersWithVariableDeclarationArgument[0]))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private class LocalVariableDeclarationVisitor : OperationWalker
+        {
+            private readonly CancellationToken _cancellationToken;
+            private readonly HashSet<ISymbol> _allSymbols;
+            private LocalVariableDeclarationVisitor(CancellationToken cancellationToken)
+            {
+                _cancellationToken = cancellationToken;
+                _allSymbols = new HashSet<ISymbol>();
+            }
+
+            public static ImmutableHashSet<ISymbol> GetAllSymbols(
+                SemanticModel semanticModel,
+                TMethodDeclarationSyntax methodDeclarationSyntax,
+                CancellationToken cancellationToken)
+            {
+                var visitor = new LocalVariableDeclarationVisitor(cancellationToken);
+                var operation = semanticModel.GetOperation(methodDeclarationSyntax, cancellationToken);
+                if (operation != null)
+                {
+                    visitor.Visit(operation);
+                }
+
+                return visitor._allSymbols.ToImmutableHashSet();
+            }
+
+            public override void Visit(IOperation operation)
+            {
+                _cancellationToken.ThrowIfCancellationRequested();
+                if (operation is IVariableDeclaratorOperation variableDeclarationOperation)
+                {
+                    _allSymbols.Add(variableDeclarationOperation.Symbol);
+                }
+
+                if (operation is ILocalReferenceOperation localReferenceOperation
+                    && localReferenceOperation.IsDeclaration)
+                {
+                    _allSymbols.Add(localReferenceOperation.Local);
+                }
+
+                base.Visit(operation);
```

Comment:
this will walk into lambdas.  is that what you want?  if so, doc it.

Reply:
no, it is not wanted. Add the check to stop visiting lamba